pulsar-transaction/coordinator/src/main/java/org/apache/pulsar/transaction/coordinator/impl/TxnLogBufferedWriterMetricsStats.java

```diff
@@ -0,0 +1,234 @@
+        pulsarBatchedLogTriggeringCountByDelayTime.remove(labelValues);
+        pulsarBatchedLogTriggeringCountByForce.remove(labelValues);
+    }
+
+    public TxnLogBufferedWriterMetricsStats(String metricsPrefix, String[] labelNames, String[] labelValues,
+                                              CollectorRegistry registry){
+        this.metricsPrefix = metricsPrefix;
+        this.labelNames = labelNames.clone();
+        this.labelValues = labelValues.clone();
+
+        String pulsarBatchedLogRecordsCountPerEntryName =
+                String.format("%s_batched_log_records_count_per_entry", metricsPrefix);
+        pulsarBatchedLogRecordsCountPerEntry = (Histogram) COLLECTOR_CACHE.computeIfAbsent(
+                pulsarBatchedLogRecordsCountPerEntryName,
+                k -> new Histogram.Builder()
+                            .name(pulsarBatchedLogRecordsCountPerEntryName)
+                            .labelNames(labelNames)
+                            .help("A metrics for how many records in per batch")
+                            .buckets(RECORD_COUNT_PER_ENTRY_BUCKETS)
+                            .register(registry));
+
+        pulsarBatchedLogRecordsCountPerEntryChild = pulsarBatchedLogRecordsCountPerEntry.labels(labelValues);
+
+        String pulsarBatchedLogEntrySizeBytesName = String.format("%s_batched_log_entry_size_bytes", metricsPrefix);
+        pulsarBatchedLogEntrySizeBytes = (Histogram) COLLECTOR_CACHE.computeIfAbsent(
+                pulsarBatchedLogEntrySizeBytesName,
+                k -> new Histogram.Builder()
+                        .name(pulsarBatchedLogEntrySizeBytesName)
+                        .labelNames(labelNames)
+                        .help("A metrics for how many bytes in per batch")
+                        .buckets(BYTES_SIZE_PER_ENTRY_BUCKETS)
+                        .register(registry));
+        pulsarBatchedLogEntrySizeBytesChild = pulsarBatchedLogEntrySizeBytes.labels(labelValues);
+
+        String pulsarBatchedLogOldestRecordDelayTimeSecondsName =
+                String.format("%s_batched_log_oldest_record_delay_time_seconds", metricsPrefix);
+        pulsarBatchedLogOldestRecordDelayTimeSeconds = (Histogram) COLLECTOR_CACHE.computeIfAbsent(
+                pulsarBatchedLogOldestRecordDelayTimeSecondsName,
+                k -> new Histogram.Builder()
+                        .name(pulsarBatchedLogOldestRecordDelayTimeSecondsName)
+                        .labelNames(labelNames)
+                        .help("A metrics for the max latency in per batch")
+                        .buckets(MAX_DELAY_TIME_BUCKETS)
+                        .register(registry));
+        pulsarBatchedLogOldestRecordDelayTimeSecondsChild =
+                pulsarBatchedLogOldestRecordDelayTimeSeconds.labels(labelValues);
+
+        String pulsarBatchedLogTriggeringCountByRecordsName =
+                String.format("%s_batched_log_triggering_count_by_records", metricsPrefix);
```

Comment:
`bufferedwriter_flush_trigger_max_records`

Reply:
I have renamed this metrics name suffix, thanks for suggestions :)